packages/commonwealth/client/scripts/views/components/UserTrainingSlider/UserTrainingSlider.tsx

```diff
@@ -0,0 +1,299 @@
+      navigate(`/dashboard/for-you`, {}, null);
+    pageName === UserTrainingCardTypes.CreateContent &&
+      navigate(`/dashboard/for-you`, {}, null);
+    pageName === UserTrainingCardTypes.FinishProfile &&
+      navigate(`/profile/edit`, {}, null);
+    pageName === UserTrainingCardTypes.ExploreCommunities &&
+      navigate(`/communities`, {}, null);
+  };
+
+  useEffect(() => {
+    if (!isLoggedIn && cardTempMarkedAsCompleted.length > 0) {
+      clearCardsTempMarkedAsCompleted();
+    }
+  }, [isLoggedIn, cardTempMarkedAsCompleted, clearCardsTempMarkedAsCompleted]);
+
+  useEffect(() => {
+    if (isLoggedIn && !isLoadingProfile && profile && profileId) {
+      // if user has already given any upvotes, then hide `give-upvote` card
+      if (profile?.totalUpvotes > 0) {
+        setShouldHideTrainingCardsPermanently(
+          profileId,
+          UserTrainingCardTypes.GiveUpvote,
+        );
+      }
+
+      // if user has any social links or email, then hide `finish-profile` card
+      const hasSocialLinks =
+        (profile?.profile?.socials || []).filter(
+          (link) => link.trim().length > 0,
+        )?.length > 0;
+      if (hasSocialLinks || app?.user?.email) {
+        setShouldHideTrainingCardsPermanently(
+          profileId,
+          UserTrainingCardTypes.FinishProfile,
+        );
+      }
+
+      // if user has created any comment/thread, then hide `create-content` card
+      if (profile?.comments?.length > 0 || profile?.threads?.length > 0) {
+        setShouldHideTrainingCardsPermanently(
+          profileId,
+          UserTrainingCardTypes.CreateContent,
+        );
+      }
+    }
+  }, [
+    isLoggedIn,
+    isLoadingProfile,
+    profile,
+    profileId,
+    shouldHideTrainingCardsPermanently,
+    setShouldHideTrainingCardsPermanently,
+  ]);
+
+  if (
+    !isLoggedIn ||
+    isLoadingProfile ||
+    (shouldHideTrainingCardsPermanently?.[profileId]?.length === 4 &&
+      cardTempMarkedAsCompleted.length === 0)
+  ) {
+    return;
+  }
+
+  const isGiveUpvoteCardVisible =
+    cardTempMarkedAsCompleted.includes(UserTrainingCardTypes.GiveUpvote) ||
+    !shouldHideTrainingCardsPermanently?.[profileId]?.includes(
+      UserTrainingCardTypes.GiveUpvote,
+    );
+  const isCreateContentCardVisible =
+    cardTempMarkedAsCompleted.includes(UserTrainingCardTypes.CreateContent) ||
+    !shouldHideTrainingCardsPermanently?.[profileId]?.includes(
+      UserTrainingCardTypes.CreateContent,
+    );
+  const isFinishProfileCardVisible =
+    cardTempMarkedAsCompleted.includes(UserTrainingCardTypes.FinishProfile) ||
+    !shouldHideTrainingCardsPermanently?.[profileId]?.includes(
+      UserTrainingCardTypes.FinishProfile,
+    );
+  const isExploreCommunitiesCardVisible =
+    cardTempMarkedAsCompleted.includes(
+      UserTrainingCardTypes.ExploreCommunities,
+    ) ||
+    !shouldHideTrainingCardsPermanently?.[profileId]?.includes(
+      UserTrainingCardTypes.ExploreCommunities,
```

Comment:
how about creating some small util, where you would pass card type and it would do all the checks and return the boolean?